components/autofill/src/db/addresses.rs

```diff
@@ -97,9 +97,13 @@ pub(crate) fn get_address(conn: &Connection, guid: &Guid) -> Result<InternalAddr
         WHERE guid = :guid",
         common_cols = ADDRESS_COMMON_COLS
     );
-
-    let address = tx.query_row(&sql, &[guid], |row| Ok(InternalAddress::from_row(row)?))?;
-
+    let result = tx.query_row(&sql, &[guid], |row| Ok(InternalAddress::from_row(row)?));
+    let address = match result {
+        Err(rusqlite::Error::QueryReturnedNoRows) => {
+            return Err(Error::NoSuchRecord(guid.to_string()));
```

Comment:
You can abbreviate this line as follows:

```
let address = match result {
        Err(rusqlite::Error::QueryReturnedNoRows) => Err(Error::NoSuchRecord(guid.to_string()))?,
        _ => result?,
    };
```

Reply:
Same for credit cards

Reply:
I wanted to use this one, but I think that was the source of the clippy error. I turned it back to a regular return, tell me if there's a method that's both compact and approved by clippy.

Reply:
```
    let address = conn
        .query_row(&sql, &[guid], |row| Ok(InternalAddress::from_row(row)?))
        .map_err(|e| match e {
            rusqlite::Error::QueryReturnedNoRows => Error::NoSuchRecord(guid.to_string()),
            e => e.into(),
        })?;
    Ok(address)

```
Opinions on Rust are very subjective, especially for newcomers - I remember my brain hurting alot with chained `.map()` calls, so whether the code above is *better* is probably contested. TBH, I'm fine either way, so if you do happen to think it's better then feel free to take it, but if you think your existing code is clearer feel free to keep that too.

Reply:
Thanks for all the help with this.  It feels kind of silly to go back and forth on such a small amount of code, but it's helpful for me to learn more rust idioms.  This version seems great to me, because map_err() describes exactly what we're doing.